src/documentation/runners/dataflow.md

```diff
@@ -4,6 +4,118 @@ title: "Cloud Dataflow Runner"
+
+The [Beam Capability Matrix]({{ site.baseurl }}/documentation/runners/capability-matrix/) documents the supported capabilities of the Cloud Dataflow runner.
+
+## Cloud Dataflow runner prerequisites and setup
+To use the Cloud Dataflow runner, you must complete the following setup:
+
+1. Select or create a Google Cloud Platform Console project.
+
+2. Enable billing for your project.
+
+3. Enable APIs: Cloud Dataflow, Compute Engine, Cloud Logging, Cloud Storage, Cloud Storage JSON, BigQuery, Cloud Pub/Sub, and Cloud Datastore.
+
+4. Install the Cloud SDK.
+
+5. Create a Cloud Storage bucket.
+    * In the Cloud Platform Console, go to the Cloud Storage browser.
+    * Click **Create bucket**.
+    * In the **Create bucket** dialog, specify the following attributes:
+      * _Name_: A unique bucket name. Do not include sensitive information in the bucket name, as the bucket namespace is global and publicly visible.
+      * _Storage class_: Multi-Regional
+      * _Location_:  Choose your desired location
+    * Click **Create**.
+
+For more information, see the *Before you begin* section of the [Cloud Dataflow quickstarts](https://cloud.google.com/dataflow/docs/quickstarts).
+
+### Specify your dependency
+
+You must specify your dependency on the Cloud Dataflow runner.
+
+For example, if you are using Maven for development and want to use the SDK for Java with [DataflowRunner]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/runners/dataflow/DataflowRunner.html), add the following dependency to your `pom.xml` file:
+
+```
+<dependency>
+  <groupId>org.apache.beam</groupId>
+  <artifactId>beam-runners-google-cloud-dataflow-java</artifactId>
+  <version>0.3.0-incubating</version>
+  <scope>runtime</scope>
+</dependency>
+```
+
+### Authentication
+
+Before running your pipeline, you must authenticate with the Google Cloud Platform. Run the following command to get [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials).
+
+```
+gcloud auth application-default login
+```
+
+## Pipeline options for the Cloud Dataflow runner
+
+When executing your pipeline from the command-line, set these pipeline options.
+
+<table class="table table-bordered">
+<tr>
+  <th>Field</th>
+  <th>Description</th>
+  <th>Default Value</th>
+</tr>
+<tr>
+  <td><code>runner</code></td>
+  <td>The pipeline runner to use. This option allows you to determine the pipeline runner at runtime.</td>
+  <td>Set to <code>dataflow</code> to run on the Cloud Dataflow Service.</td>
+</tr>
+<tr>
+  <td><code>project</code></td>
+  <td>The project ID for your Google Cloud Project.</td>
+  <td>If not set, defaults to the default project of the current user.</td>
+</tr>
+<tr>
+  <td><code>streaming</code></td>
+  <td>Whether streaming mode is enabled or disabled; <code>true</code> if enabled.</td>
```

Comment:
Done. I followed the style of the programming guide, though it looks a bit strange because the code block font is so different from the non-code font. Other possible options would be "PCollection objects" or just "collections", if the visual ickyness is too much.